agent/logging.go

```diff
@@ -87,33 +96,99 @@ func (a *Agent) makeLoggerProducer(ctx context.Context, c *model.LoggerConfig, t
 			LogkeeperURL:      a.opts.LogkeeperURL,
 			LogkeeperBuilder:  task.Id,
 			LogkeeperBuildNum: task.Execution,
-			Sender:            model.LogSender(agentConfig.Type),
+			Sender:            agentConfig.Type,
 			SplunkServerURL:   agentConfig.SplunkServer,
 			SplunkToken:       agentConfig.SplunkToken,
-			Filepath:          fmt.Sprintf("%s/agent.log", path),
+			Filepath:          fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, agentLogs),
 		})
 	}
 	for _, systemConfig := range c.System {
 		config.System = append(config.System, client.LogOpts{
 			LogkeeperURL:      a.opts.LogkeeperURL,
 			LogkeeperBuilder:  task.Id,
 			LogkeeperBuildNum: task.Execution,
-			Sender:            model.LogSender(systemConfig.Type),
+			Sender:            systemConfig.Type,
 			SplunkServerURL:   systemConfig.SplunkServer,
 			SplunkToken:       systemConfig.SplunkToken,
-			Filepath:          fmt.Sprintf("%s/system.log", path),
+			Filepath:          fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, systemLogs),
 		})
 	}
 	for _, taskConfig := range c.Task {
 		config.Task = append(config.Task, client.LogOpts{
 			LogkeeperURL:      a.opts.LogkeeperURL,
 			LogkeeperBuilder:  task.Id,
 			LogkeeperBuildNum: task.Execution,
-			Sender:            model.LogSender(taskConfig.Type),
+			Sender:            taskConfig.Type,
 			SplunkServerURL:   taskConfig.SplunkServer,
 			SplunkToken:       taskConfig.SplunkToken,
-			Filepath:          fmt.Sprintf("%s/task.log", path),
+			Filepath:          fmt.Sprintf("%s/%s", a.opts.WorkingDirectory, taskLogs),
 		})
 	}
 	return a.comm.GetLoggerProducer(ctx, td, &config)
 }
+
+type s3UploadOpts struct {
```

Comment:
this feels like there's overlap with the other s3 options struct.

are we exposing configuration that sends logs to more than one bucket? or allow projects to specify different buckets for different logs. Do we even want that?

Reply:
deleted